### DISCUSSION TITLE
How can I enable feature for all tenant

### DISCUSSION THREAD

**Discussion by sbiaudet** [OrchardCMS/OrchardCore#13186] [General]
I've a question to enable a feature for all tenant. I've an existing multi-tenant solution and I wan't to move to a distributed architecture. 

Is it possible to enable features OrchardCore.Medias.Azure and OrchardCore.DataProtection.Azure via appsettings ?

We have thinking other solutions (SqlUpdate, Migration, ...). What is the best solution ?

Regards

Sébastien

**Comment by jtkech** [comment]
From you web app, `AddOrchardCms()` returns an `OrchardCoreBuilder` on which we can chain other extension helpers, we already use `AddSetupFeatures()` to enable feature(s) but only during a setup, we have other helpers e.g. to enable only one feature, another to enable one or multiple features.

    builder.Services
        .AddOrchardCms()
        .EnableFeature("FeaturA1")
        .AddGlobalFeatures("FeatureB1", "FeatureB2")
        .AddSetupFeatures("OrchardCore.AutoSetup");

Hmm yes, from memory we also have a `Features` config field that you can use under the `OrchardCore` section of any config source e.g. `appsettings.json` to define an array of features `"Feaures": [ "Feature1", "Feature2" ]`.

Then in a distributed env we have a set of features that you may need or not, for example to keep in sync admin settings without having to restart all instances by using a distributed cache e.g. `OC.Redis.Cache` but that needs a Redis instance.

**Reply by sbiaudet** [reply]
Thank you !! 

I have been following your work on distributed environments. I can see what I have to do. If need be I'll let you know.
